### PR TITLE
quiche: update tar

### PR DESCRIPTION
**Pull request by danzh2010**
Signed-off-by: Dan Zhang <danzh@google.com>

Update to commit b2b8ff25f5a565324b93411ca29c3403ccbca969

Risk Level: low
Testing: existing tests pass

### CODE REVIEW

**Comment by danzh2010** [comment]
This should unblock https://github.com/envoyproxy/envoy/pull/11941

File: test/extensions/quic_listeners/quiche/active_quic_listener_test.cc [hunk 1]
```diff
@@ -84,9 +84,9 @@ class ActiveQuicListenerTest : public QuicMultiVersionTest {
             return quic::CurrentSupportedVersionsWithQuicCrypto();
           }
           bool use_http3 = GetParam().second == QuicVersionType::Iquic;
-          SetQuicReloadableFlag(quic_enable_version_draft_28, use_http3);
-          SetQuicReloadableFlag(quic_enable_version_draft_27, use_http3);
-          SetQuicReloadableFlag(quic_enable_version_draft_25_v3, use_http3);
+          SetQuicReloadableFlag(quic_enable_version_draft_29, use_http3);
+          SetQuicReloadableFlag(quic_disable_version_draft_27, use_http3);
+          SetQuicReloadableFlag(quic_disable_version_draft_25, use_http3);
```
Comment (wu-bin) [inline test/extensions/quic_listeners/quiche/active_quic_listener_test.cc:89]: Note the two flags are quic_disable_*,  it seems you meant !use_http3.

File: test/extensions/quic_listeners/quiche/envoy_quic_client_session_test.cc [hunk 2]
```diff
@@ -95,9 +95,9 @@ class EnvoyQuicClientSessionTest : public testing::TestWithParam<bool> {
       : api_(Api::createApiForTest(time_system_)),
         dispatcher_(api_->allocateDispatcher("test_thread")), connection_helper_(*dispatcher_),
         alarm_factory_(*dispatcher_, *connection_helper_.GetClock()), quic_version_([]() {
-          SetQuicReloadableFlag(quic_enable_version_draft_28, GetParam());
-          SetQuicReloadableFlag(quic_enable_version_draft_27, GetParam());
-          SetQuicReloadableFlag(quic_enable_version_draft_25_v3, GetParam());
+          SetQuicReloadableFlag(quic_enable_version_draft_29, GetParam());
+          SetQuicReloadableFlag(quic_disable_version_draft_27, GetParam());
+          SetQuicReloadableFlag(quic_disable_version_draft_25, GetParam());
```
Comment (wu-bin) [inline test/extensions/quic_listeners/quiche/envoy_quic_client_session_test.cc:100]: Ditto

File: test/extensions/quic_listeners/quiche/envoy_quic_dispatcher_test.cc [hunk 3]
```diff
@@ -63,9 +63,9 @@ class EnvoyQuicDispatcherTest : public QuicMultiVersionTest,
             return quic::CurrentSupportedVersionsWithQuicCrypto();
           }
           bool use_http3 = GetParam().second == QuicVersionType::Iquic;
-          SetQuicReloadableFlag(quic_enable_version_draft_28, use_http3);
-          SetQuicReloadableFlag(quic_enable_version_draft_27, use_http3);
-          SetQuicReloadableFlag(quic_enable_version_draft_25_v3, use_http3);
+          SetQuicReloadableFlag(quic_enable_version_draft_29, use_http3);
+          SetQuicReloadableFlag(quic_disable_version_draft_27, use_http3);
+          SetQuicReloadableFlag(quic_disable_version_draft_25, use_http3);
```
Comment (wu-bin) [inline test/extensions/quic_listeners/quiche/envoy_quic_dispatcher_test.cc:68]: Ditto

File: test/extensions/quic_listeners/quiche/integration/quic_http_integration_test.cc [hunk 4]
```diff
@@ -290,13 +290,12 @@ TEST_P(QuicHttpIntegrationTest, MultipleQuicListenersWithBPF) {
   set_reuse_port_ = true;
   initialize();
   std::vector<IntegrationCodecClientPtr> codec_clients;
-  quic::QuicCryptoClientConfig::CachedState* cached = crypto_config_.LookupOrCreate(server_id_);
   for (size_t i = 1; i <= concurrency_; ++i) {
     // The BPF filter looks at the 1st word of connection id in the packet
     // header. And currently all QUIC versions support 8 bytes connection id. So
     // create connections with the first 4 bytes of connection id different from each
     // other so they should be evenly distributed.
```
Comment (wu-bin) [inline test/extensions/quic_listeners/quiche/integration/quic_http_integration_test.cc:297]: btw, we have change the BPF internally to inspect the 4 bytes starting from byte 1 (0-based). You may want to do the same here.

Comment (danzh2010) [reply test/extensions/quic_listeners/quiche/integration/quic_http_integration_test.cc:297]: ACK. I will make another change accordingly. What's the reason behind the change?

File: test/extensions/quic_listeners/quiche/envoy_quic_dispatcher_test.cc (hunk 3 above)
Comment (danzh2010) [reply test/extensions/quic_listeners/quiche/envoy_quic_dispatcher_test.cc]: done

File: test/extensions/quic_listeners/quiche/envoy_quic_client_session_test.cc (hunk 2 above)
Comment (danzh2010) [reply test/extensions/quic_listeners/quiche/envoy_quic_client_session_test.cc]: done

File: test/extensions/quic_listeners/quiche/active_quic_listener_test.cc (hunk 1 above)
Comment (danzh2010) [reply test/extensions/quic_listeners/quiche/active_quic_listener_test.cc]: done

**Comment by moderation** [comment]
@danzh2010 despite https://github.com/envoyproxy/envoy/blob/master/source/extensions/extensions_build_config.bzl#L165 there is still no way to enable QUIC yet right?

**Pull request merged**